examples/NSIDC/cmr_auth.py

```diff
@@ -0,0 +1,66 @@
+from copy import deepcopy
+from typing import Dict, Any
+
+from requests import session, exceptions
+from requests.auth import HTTPBasicAuth
+
+
+class CMRAuth(object):
+    """
+    base class to perform authentication operations for NASA CMR
+    """
+    CMR_OPS = "https://cmr.earthdata.nasa.gov/legacy-services/rest/tokens"
+    CMR_UAT = "https://cmr.uat.earthdata.nasa.gov/legacy-services/rest/tokens"
+    CMR_SIT = "https://cmr.sit.earthdata.nasa.gov/legacy-services/rest/tokens"
+
+    def __init__(self, username: str, password: str, environment_url: str=CMR_OPS):
+        self.session = session()
+        if username is not None and password is not None:
+            _TOKEN_DATA = ('<token>'
+                             '<username>%s</username>'
+                             '<password>%s</password>'
+                             '<client_id>CMR Client</client_id>'
+                             '<user_ip_address>%s</user_ip_address>'
+                           '</token>'
+                           )
+            my_ip = self.session.get('https://ipinfo.io/ip').text.strip()
+            self.auth = HTTPBasicAuth(username, password)
+            # This token is valid for up to 3 months after is issued.
```

Comment:
Ah hah! Good to know. I recommend putting this information in the notebook so people will see it in the cookbook :)